verilog/tools/extractor/verilog_extractor_types.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_EXTRACTOR_VERILOG_EXTRACTOR_TYPES_H
+#define VERIBLE_VERILOG_TOOLS_EXTRACTOR_VERILOG_EXTRACTOR_TYPES_H
+
+#include <iosfwd>
+
+enum class Type {
+    // BEGIN GENERATE -- do not delete
+    NoType,
+    File,
+    Module,
+    ModuleHeader,
+    ModuleBody,
+    ModuleEnd,
```

Comment:
Not clear to me what the role of these is.
Are these really just implementation details that don't need to be exposed?

Reply:
These are like the Tags used in CST.
This isn't yet completed and contains unused values 'cos it's still changing.